backend/utils/arrays.go

```diff
@@ -0,0 +1,194 @@
+package utils
+
+import (
+	"reflect"
+	"sort"
+	"strconv"
+	"strings"
+	"time"
+)
+
+// Element Array elements as object
+type Element interface{}
+
+// WrapArray Array elements to object
+func WrapArray(arrayData interface{}) map[string]Element {
+	return WrapArrayWithElemName(arrayData, "data")
+}
+
+// WrapArrayWithElemName Array elements to object with field name.
+func WrapArrayWithElemName(arrayData interface{}, elemName string) map[string]Element {
+	jsonData := make(map[string]Element)
+	dataSize := getArraySize(arrayData)
+	if arrayData == nil || dataSize <= 0 {
+		jsonData[elemName] = make([]Element, 0)
+	} else {
+		jsonData[elemName] = arrayData
+	}
+
+	return jsonData
+}
+
+func getArraySize(data interface{}) int {
+	if data == nil {
+		return 0
+	}
+	switch reflect.TypeOf(data).Kind() {
+	case reflect.Slice:
+		return reflect.ValueOf(data).Len()
+	default:
+		return -1
+	}
+}
+
+// IndexOfInt returns index of value in the array.
+func IndexOfInt(arrayData []int, val int) int {
+	for i, v := range arrayData {
+		if v == val {
+			return i
+		}
+	}
+	return -1
+}
+
+// ExistsInt returns the value exists in the array.
+func ExistsInt(arrayData []int, val int) bool {
+	for _, v := range arrayData {
+		if v == val {
+			return true
+		}
+	}
+	return false
+}
+
+// UniqInt returns duplicate value in the array.
+func UniqInt(arrayData []int) []int {
+	results := make([]int, 0, len(arrayData))
+	encounters := map[int]bool{}
+	for _, v := range arrayData {
+		if !encounters[v] {
+			results = append(results, v)
+			encounters[v] = true
+		}
+	}
+	return results
+}
+
+// ExistsString returns the value exists in the array.
+func ExistsString(arrayData []string, val string) bool {
+	for _, v := range arrayData {
+		if v == val {
+			return true
+		}
+	}
+	return false
+}
+
+// IntArrayToString 数値の配列を delimiter で結合した文字列で返却する.
```

Comment:
are there any needs for Japanese comments?😂

Reply:
The backend is base on my former project, just forgot remove Japanese comments.Never mind 😂